src/main/java/mate/academy/dao/impl/BookDaoImpl.java

```diff
@@ -0,0 +1,135 @@
+        }
+        return Optional.empty();
+    }
+
+
+    @Override
+    public List<Book> findAll() {
+        String selectRequest = "SELECT id, title, price "
+                + "FROM books "
+                + "WHERE is_deleted = 0;";
+        try (Connection connection = ConnectionUtil.getConnection();
+             PreparedStatement getManufacturerStatements =
+                     connection.prepareStatement(selectRequest)) {
+            ResultSet resultSet = getManufacturerStatements.executeQuery();
+            List<Book> bookList = new ArrayList<>();
+            while (resultSet.next()) {
+                Long id = resultSet.getLong("id");
+                String title = resultSet.getString("title");
+                BigDecimal price = resultSet.getBigDecimal("price");
+                bookList.add(getBook(id, title, price));
+            }
+            return bookList;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't select all books from DB", e);
+        }
+    }
+
+    @Override
+    public Book update(Book book) {
+        String updateRequest = "UPDATE books "
+                + "SET title = ?, price = ? "
+                + "WHERE id = ? AND is_deleted = 0";
+        try (Connection connection = ConnectionUtil.getConnection();
+             PreparedStatement updateManufacturerStatements =
+                     connection.prepareStatement(updateRequest)) {
+            updateManufacturerStatements.setString(1, book.getTitle());
+            updateManufacturerStatements.setBigDecimal(2, book.getPrice());
+            updateManufacturerStatements.setLong(3, book.getId());
+            int rows = updateManufacturerStatements.executeUpdate();
+            if (rows != 1) {
+                throw new DataProcessingException("Can't update book "
+                        + book, new SQLException());
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update book "
+                    + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        String updateRequest = "UPDATE books "
+                + "SET is_deleted = 1 "
+                + "WHERE id = ? AND is_deleted = 0";
+        try (Connection connection = ConnectionUtil.getConnection();
+             PreparedStatement updateManufacturerStatements =
+                     connection.prepareStatement(updateRequest)) {
+            updateManufacturerStatements.setLong(1, id);
+            int rows = updateManufacturerStatements.executeUpdate();
+            if (rows == 1) {
+                return true;
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete book with ID "
+                    + id, e);
+        }
+        return false;
+    }
+
+    private Book getBook(Long id, String title, BigDecimal price) {
```

Comment:
You can pass ResultSet here instead of id, title, and price.
And parse it directly in the method